frontstage/controllers/party_controller.py

```diff
@@ -552,15 +551,12 @@ def get_surveys_listed_against_party_and_business_id(business_id, party_id):
     """
     returns list of surveys associated with a business id and respondent
     :param business_id: business id
-    :param business_id: party id (respondent party id)
+    :param party_id: The respondent's party id
     :return: list of surveys
     :rtype: list
     """
-    enrolment_data = list(get_respondent_enrolments(party_id))
-    survey_ids = set()
-    for enrolment in enrolment_data:
-        if enrolment['business_id'] == business_id:
-            survey_ids.add(enrolment['survey_id'])
+    enrolment_data = get_respondent_enrolments(party_id)
+    survey_ids = {enrolment['survey_id'] for enrolment in enrolment_data if enrolment['business_id'] == business_id}
```

Comment:
@insacuri - What happens in duplicate business?

Reply:
Functionally this is the same as what was there before. Using curly braces makes it a set comprehension.  Like a set, if you add the same thing twice, then duplicates won't be added.